lib/maths/analytics/CBoostedTreeHyperparameters.cc

```diff
@@ -46,6 +46,32 @@ CBoostedTreeHyperparameters::CBoostedTreeHyperparameters() {
     this->initializeTunableHyperparameters();
 }
 
+CBoostedTreeHyperparameters::CBoostedTreeHyperparameters(const CBoostedTreeHyperparameters& other) {
+    *this = other;
+}
+
+CBoostedTreeHyperparameters& CBoostedTreeHyperparameters::
+operator=(const CBoostedTreeHyperparameters& other) {
```

Comment:
I'm not keen on having actual copying only copying part of the object. I think it would be better to have a named method to make it clear this isn't a real copy. Maybe something like:
```
void copyCurrentValues(const CBoostedTreeHyperparameters& other);
```

The same comment applies to the copy constructor.

Reply:
Alright, I introduced the private method `copyRegularizationParameters()` and removed the public constructor and assignment operators.